libstorage/src/Volume.cc

```diff
@@ -2390,30 +2384,35 @@ int Volume::doCryptsetup(bool readonly)
 	        (format || (isTmpCryptMp(mp)&&crypt_pwd.empty()) ||
 	         (encryption!=ENC_NONE&&orig_crypt_pwd!=crypt_pwd)) )
 		{
-		string cmdline = getCryptsetupCmd( encryption, dmcrypt_dev, mp, fname, true,
-						   crypt_pwd.empty() );
-		if( !cmdline.empty() )
+                SystemCmd * cryptSetupCmd = createCryptSetupCmd( encryption, dmcrypt_dev, mp, crypt_pwd, true );
+                if( cryptSetupCmd )
 		    {
-		    cmd.execute( cmdline );
-		    if( cmd.retcode()!=0 )
+                    cryptSetupCmd->execute();
+                    if( cryptSetupCmd->retcode() != 0 )
 			ret = VOLUME_CRYPTFORMAT_FAILED;
+
+                    delete cryptSetupCmd;
+                    cryptSetupCmd = 0;
+
 		    if( ret==0 && mp=="swap" && crypt_pwd.empty() && !format )
-			cmd.execute("/sbin/mkswap " + quote(dmcrypt_dev));
+			SystemCmd mkswapCmd("/sbin/mkswap " + quote(dmcrypt_dev));
 		    }
 		}
 	    if( ret==0 && (!isTmpCryptMp(mp)||!crypt_pwd.empty()) )
 		{
-		string cmdline = getCryptsetupCmd( encryption, dmcrypt_dev, mp, fname, false );
-		if( !cmdline.empty() )
+		SystemCmd * cryptSetupCmd = createCryptSetupCmd( encryption, dmcrypt_dev, mp, crypt_pwd, false );
```

Comment:
This patterns repeats several times... But given this is just a backport I'm fine with that.